### DISCUSSION TITLE
Compile windows MSR driver

### DISCUSSION THREAD

**Discussion by OualidBougzime** [opcm/pcm#372] [Q&A]
Hi!

For the project, I can't compile the msr.sys, I've already installed the DDK kit but I can't find out where this file is and what I have to do exactly to compile it!

Thanks for your help!

**Comment by rdementi** [comment]
the project file for the driver is in WinMSRDriver source directory.

**Reply by OualidBougzime** [reply]
I can't run the source directory, there is a problem. Which file should I run?

**Comment by ogbrugge** [comment]
Might it be that you need to run cmake first and it is now auto-generated just like the other project files? I ran into this after the cmake conversion, not sure though if the driver project file was also affected by that change

**Reply by opcm** [reply]
this one is not auto-generated

**Comment by twillhal** [comment]
Compiling and signing a driver is usually not straight forward. Have you considered using one of the precompiled drivers that are mentioned in the [Windows howto](https://github.com/opcm/pcm/blob/master/doc/WINDOWS_HOWTO.md)?

**Reply by OualidBougzime** [reply]
Thanks for your response!

Yes absolutely, I used the MSR Driver but I can't sign it! There is a problem. How did you sign it?

**Comment by ogbrugge** [comment]
https://docs.microsoft.com/en-us/windows-hardware/drivers/install/windows-driver-signing-tutorial

Does this document help? The part at the bottom regarding self signing, not the MSFT WHQL signing. You may still need to put your windows in a special mode where it accepts self signed drivers, not sure if that documentation mentions that.
